### PR TITLE
send password to first connected server

### PR DESCRIPTION
**Pull request by ZombieToad**
https://github.com/ddnet/ddnet/pull/1149#issuecomment-397538132

fixes issues with connecting to servers with reserved slots

### CODE REVIEW

**Comment by Learath2** [comment]
Should probably rename this command, other then that looks fine, yet still feels horribly hacky. Maybe a CTRL+Enter or CTRL+DoubleClick should send the password (`menus_browser.cpp:L483`). It would prevent accidental leakage and let people connect to servers with reserve slots.

**Comment by def-** [comment]
But it's intentional not to send the password initially. Otherwise the server owner can pretend to have a password and gets people's secret passwords.

**Comment by ZombieToad** [comment]
misworded it. @def- it sends password to first connected server after you enter a password in f1

**Comment by def-** [comment]
but that also seems bad. I automatically set a password when starting my client (something like `./DDNet password foo`) and if I then entered a server that didn't belong to me, they could immediately figure out my password. Right now I still get a screen where I have to press enter to confirm that I want to send them the password.

**Comment by Learath2** [comment]
@def- how would you feel about what I proposed instead?

**Comment by def-** [comment]
How would people know that ctrl+enter does something? seems undiscoverable

**Comment by heinrich5991** [comment]
@def- Maybe you could adjust your config file to include something like `bind f10 password abc`, so you just press the key to send the password to the next server?

File: src/engine/client/client.cpp [hunk 1]
```diff
@@ -3402,6 +3423,7 @@ void CClient::RegisterCommands()
 	m_pConsole->Register("disconnect", "", CFGFLAG_CLIENT, Con_Disconnect, this, "Disconnect from the server");
 	m_pConsole->Register("ping", "", CFGFLAG_CLIENT, Con_Ping, this, "Ping the current server");
 	m_pConsole->Register("screenshot", "", CFGFLAG_CLIENT, Con_Screenshot, this, "Take a screenshot");
+	m_pConsole->Register("password", "?r[password]", CFGFLAG_CLIENT, Con_Password, this, "Password to the server");
```
Comment (heinrich5991) [inline src/engine/client/client.cpp]: There's a `m_pConsole->Chain` that more closely resembles on what you want to do (it allows you to add extra actions to a console command).

**Comment by def-** [comment]
But after you joined a server with a password that password is set. Next time I join another hostile server my previous password will be auto-used.

**Comment by heinrich5991** [comment]
True. It should only be sent to the next server you join.

**Comment by ZombieToad** [comment]
it does only send 1 time after you type password?

**Comment by def-** [comment]
Meh, ok.

**Comment by heinrich5991** [comment]
@ZombieToad Could you change the the code to use `m_pConsole->Chain`?

**Comment by ZombieToad** [comment]
i made changes @heinrich5991 wanted and i think i fix problem @def-  have  `./DDNet "password foo"` won't send password

**Comment by heinrich5991** [comment]
Thanks for also finding a way to fix deen's concern!

bors r+

**Pull request merged**